### DISCUSSION TITLE
Serialise array as map of index: items

### DISCUSSION THREAD

**Discussion by detly** [jonasbb/serde_with#206] [Features]
Maybe this is a bit niche, but I've seen it in a few APIs now. Instead of sending:

```json
{
  "deltas": [
    "update for 1",
    "update for 2",
    null,
    null,
    "update for 5"
  ]
}
```

...to save space, an API might send:

```json
{
  "deltas": {
    "1": "update for 1",
    "2": "update for 2",
    "5": "update for 5"
  }
}
```

But in Rust code you might still want to enforce the size of the array, so you have:

```rust
struct Updates {
  deltas: [Option<String>; NUM_THINGS],
}
```

It would be nice if `serde_as(...)` could express this in the same way it can handle a vector of tuples. It might require [const generics](https://rust-lang.github.io/rfcs/2000-const-generics.html), but maybe I'm wrong about that. At the moment, the only workaround I can think of is to use vanilla Serde with something like:

```rust
struct SomeNumberOfUpdates {
  #[serde(rename = "1")
  update_for_1: Option<String>,
  // ... times 100, yikes
  #[serde(rename = "100")
  update_for_100: Option<String>,
}
```

If this is already possible, excuse the spurious request, but it might make a nice next-level example in the docs?

**Comment by jonasbb** [comment]
There is nothing like that in the library so far. I haven't seen an API like that. Do you have an example? Pure curiosity.
Such a conversion would fit in this library.

I have two questions regarding your example:

1. How strict/flexible are the numeric keys? Are they 1-based or are you aware or 0-based indexing too?
2. I think writing this code for arrays might be tricky. Using a `Vec` should work. The `Vec` can then later be converted into an array. Alternatively, something like [`serde-big-array`](https://docs.rs/serde-big-array/) should work too.
